Kinvey/Kinvey/RequestFactory.swift

```diff
@@ -63,3 +63,22 @@ protocol RequestFactory {
     func buildLiveStreamUnsubscribe(streamName: String, userId: String, deviceId: String) -> HttpRequest
     
 }
+
+//Allow to set a per request
+public struct RequestConfiguration {
```

Comment:
@heyzooi this makes sense for the `customRequestHeaders` problem you're solving. As my other comments suggest, you should consider using  this requestConfig definition to encapsulate other custom parameters that are exposed on each API.